drivers/l3gxxxx/l3gxxxx.c

```diff
@@ -0,0 +1,1163 @@
+        l3gxxxx_raw_data_fifo_t raw_fifo = {};
+        int num;
+
+        /* read raw data samples from FIFO */
+        if ((num = l3gxxxx_read_raw_fifo(dev, raw_fifo)) < 0) {
+            DEBUG_DEV("reading raw data samples from FIFO failed", dev);
+            return num;
+        }
+
+        if (num == 0) {
+            DEBUG_DEV("no raw data in FIFO", dev);
+            return -L3GXXXX_ERROR_NO_NEW_DATA;
+        }
+
+        *raw = raw_fifo[num - 1];
+
+        return res;
+    }
+#endif /* IS_USED(MODULE_L3GXXXX_FIFO) */
+
+    uint8_t data[6];
+
+    /* read raw data sample and clear interrupt signal INT2/DRDY if used */
+    if (l3gxxxx_reg_read(dev, L3GXXXX_REG_OUT_X_L, data, 6)) {
+        DEBUG_DEV("could not get raw data", dev);
+        return -L3GXXXX_ERROR_RAW_DATA;
+    }
+
+    /* L3GXXXX_REG_CTRL4.BLE = 0, Data LSB @ lower address */
+    raw->x = (data[1] << 8) | data[0];
+    raw->y = (data[3] << 8) | data[2];
+    raw->z = (data[5] << 8) | data[4];
+
+    return res;
+}
+
+#if IS_USED(MODULE_L3GXXXX_FIFO)
+
+int l3gxxxx_read_fifo(const l3gxxxx_t *dev, l3gxxxx_data_fifo_t data)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(data != NULL);
+    DEBUG_DEV("", dev);
+
+    l3gxxxx_raw_data_fifo_t raw = {};
+    int num;
+
+    if ((num = l3gxxxx_read_raw_fifo(dev, raw)) < 0) {
+        DEBUG_DEV("reading raw data samples from FIFO failed", dev);
+        return num;
+    }
+
+    for (int i = 0; i < num; i++) {
+        data[i].x = (raw[i].x * L3GXXXX_SCALES[dev->params.scale]) >> 2;
+        data[i].y = (raw[i].y * L3GXXXX_SCALES[dev->params.scale]) >> 2;
+        data[i].z = (raw[i].z * L3GXXXX_SCALES[dev->params.scale]) >> 2;
+    }
+
+    return num;
+}
+
+int l3gxxxx_read_raw_fifo(const l3gxxxx_t *dev, l3gxxxx_raw_data_fifo_t raw)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(raw != NULL);
+    DEBUG_DEV("", dev);
+
+    int res = L3GXXXX_OK;
+
+    /* in bypass mode, we use l3gxxxx_read_raw to return one sample */
+    if (dev->params.fifo_mode == L3GXXXX_BYPASS) {
+        res = l3gxxxx_read_raw(dev, raw);
+        return (res == L3GXXXX_OK) ? 1 : res;
+    }
+
+    uint8_t reg;
+
+    _acquire(dev);
+
+    /* read FIFO state */
+    if (_read(dev, L3GXXXX_REG_FIFO_SRC, &reg, 1) != L3GXXXX_OK) {
+        _release(dev);
+        return res;
+    }
+
+    /* if nothing is in the FIFO, just return with 0 */
+    if (reg & L3GXXXX_FIFO_EMPTY) {
+        _release(dev);
+        return 0;
+    }
+
+    /* read samples from FIFO */
+    int samples = (reg & L3GXXXX_FIFO_FFS) + ((reg & L3GXXXX_FIFO_OVR) ? 1 : 0);
+
+    /* read samples from FIFO */
+    for (int i = 0; i < samples; i++) {
+
+        uint8_t data[6];
+
+        /* read raw data sample */
+        res |= _read(dev, L3GXXXX_REG_OUT_X_L, data, 6);
+
+        /* L3GXXXX_REG_CTRL4.BLE = 0, Data LSB @ lower address */
+        raw[i].x = (data[1] << 8) | data[0];
+        raw[i].y = (data[3] << 8) | data[2];
+        raw[i].z = (data[5] << 8) | data[4];
```

Comment:
You could also do
```suggestion
        raw[i].x = byteorder_lebuftohs(&data[0]);
        raw[i].y = byteorder_lebuftohs(&data[2]);
        raw[i].z = byteorder_lebuftohs(&data[4]);
```
in case one of these mythical big endian machines shows up one day. But then again they are pretty much extinct for all practical purpose.